releasenotes/notes/deprecate-basic-simulator-configuration-9d782925196993e9.yaml

```diff
@@ -0,0 +1,29 @@
+---
+deprecations_providers:
+  - |
+    The :meth:`.BasicSimulator.configuration` method is deprecated and will be removed in 2.0.0. 
+    This method returned a legacy ``providers.models.BackendConfiguration`` instance which is part
+    of the deprecated ``BackendV1`` model. This model has been replaced with :class:`.BackendV2`, 
+    where the constraints are stored directly in the backend instance or the underlying :class:`.Target`
+    (``backend.target``).
+
+    Here is a quick guide for accessing the most common ``BackendConfiguration`` attributes in the 
+    :class:`BackendV2` model:""
+
+      BackendV1 model (deprecated) ------------> BackendV2 model
+      ----------------------------               ---------------
+      backend.configuration().backend_name       backend.name
+      backend.configuration().backend_version    backend.backend_version
+      backend.configuration().n_qubits           backend.num_qubits
+      backend.configuration().num_qubits         backend.num_qubits
+      backend.configuration().basis_gates        backend.target.operation_names (*)
+      backend.configuration().coupling_map       backend.target.build_coupling_map()
+      backend.configuration().local              No representation
+      backend.configuration().simulator          No representation
+      backend.configuration().conditional        No representation
+      backend.configuration().open_pulse         No representation     
+      backend.configuration().memory             No representation      
+      backend.configuration().max_shots          No representation      
+
+    (*) Note that ``backend.target.operation_names`` includes ``basis_gates`` and additional 
+    non-gate instructions, in some implementations it might be necessary to filter the output.
```

Comment:
Is it worth just linking to https://github.com/Qiskit/qiskit/blob/main/qiskit/providers/__init__.py#L679 here too?

Reply:
There isn't a lot of extra information but I added it in case the migration guide gets improvements (I know there's an open issue for that).